engine/lib/new_generic_printer/new_generic_printer_api.ml

```diff
@@ -0,0 +1,92 @@
+open! Prelude
```

Comment:
Not sure if the name is the best ....
But aside from that, we need a design of this. Please add a design document (markdown, mermaind, whatever) that describes how this printer works and how it can be used.
And then document the code (with how it's implementing the design).